packages/buildessential.rb

```diff
@@ -7,7 +7,7 @@ class Buildessential < Package
 
   is_fake
 
-  depends_on 'gcc'
+  depends_on 'gcc7'
```

Comment:
You sure we're ready for this?  Bump version?

Reply:
The option gcc4-compatible is added, no need to compile the packages from source code again.

Reply:
Let's test this a bit more.  Please compile some more packages with gcc7 and push a few PRs.

Reply:
Tests results good.  Nice work @lovedheart!